src/test/kotlin/sirius/web/templates/SmartLineBasedProcessorTest.kt

```diff
@@ -9,49 +9,57 @@
 package sirius.web.templates
 
 import com.google.common.collect.Lists
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
 import sirius.kernel.BaseSpecification
+import sirius.kernel.SiriusExtension
 import sirius.web.data.LineBasedProcessor
 import sirius.web.data.NamedRowProcessor
 import sirius.web.data.SmartLineBasedProcessor
 import sirius.web.data.SmartRow
+import java.lang.Exception
 
 import java.util.function.Predicate
+import kotlin.reflect.KClass
+import kotlin.test.assertEquals
 
-class SmartLineBasedProcessorSpec extends BaseSpecification {
+@ExtendWith(SiriusExtension::class)
+class SmartLineBasedProcessorTest {
 
-    def "reading CSVs works with different column orders and aliases"() {
-        given:
-        List<SmartRow> contents1 = Lists.newArrayList()
-        List<SmartRow> contents2 = Lists.newArrayList()
-        SmartLineBasedProcessor proc1 = new SmartLineBasedProcessor()
+    @Test
+    fun `reading CSVs works with different column orders and aliases`() {
+        
+        val contents1 = Lists.newArrayList<SmartRow>()
+        val contents2 = Lists.newArrayList<SmartRow>()
```

Comment:
**Optional:** The Kotlin-way here is: `mutableListOf<SmartRow>()`. In Kotlin the developer can decide between a mutable list and an immutable list during the definition (same for sets and maps).